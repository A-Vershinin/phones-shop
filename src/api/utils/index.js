export const phonesUrl = 'http://www.mocky.io/v2/5accd2a73200005c00776564';
