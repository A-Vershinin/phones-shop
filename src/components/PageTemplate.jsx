import React, { PureComponent } from 'react';

import SideBarContainer from '../containers/SideBar/index';
import MenuContainer from '../containers/Menu/index';

class PageTemplate extends PureComponent {
	render() {
		return (
			<div className='container view-container'>
				<h1>Layout</h1>
				<MenuContainer/>
				<div className='row'>
					<SideBarContainer/>
					<div className='col-md-9'>{this.props.children}</div>
				</div>
			</div>
		);
	}

}

export default PageTemplate;
