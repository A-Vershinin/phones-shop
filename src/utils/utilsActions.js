import { random, lorem, commerce, image } from 'faker';

export const number = 1;

export const item = {
	"id": number,
	"categoryId": random.number(),
	"name": lorem.word(),
	"description": lorem.paragraph(),
	"price": commerce.price(),
	"image": image.image(),
	"cpu": lorem.words(),
	"camera": lorem.words(),
	"size": lorem.words(),
	"weight": lorem.words(),
	"display": lorem.words(),
	"battery": lorem.words(),
	"memory": lorem.words()
};

export const mockID = random.number();


export const dumpItem = {
	"id": 1,
	"name": lorem.word()
};

export const dumpPhones = [item];


export const itemPhone = {
	"id": random.uuid(),
	"categoryId": random.number(),
	"name": lorem.word(),
	"description": lorem.paragraph(),
	"price": commerce.price(),
	"image": image.image(),
	"cpu": lorem.words(),
	"camera": lorem.words(),
	"size": lorem.words(),
	"weight": lorem.words(),
	"display": lorem.words(),
	"battery": lorem.words(),
	"memory": lorem.words()
};

export const phonesArray = [itemPhone];

export const strPhone = "1";

export const itemPhoneMock = {
	id: '1',
	categoryId: '1',
	name: 'Apple iPhone 5c',
	description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta',
	price: 823,
	image: '/uploads/iphone5c-selection-hero-2013.png',
	cpu: '1.3GHz Apple A6',
	camera: '8mp (3264x2448)',
	size: '124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)',
	weight: '132 grams (4.7 ounces) with battery',
	display: '4.0 326 pixel density',
	battery: '1480 mAh',
	memory: '16GB, 32GB and RAM 1 GB'
}
