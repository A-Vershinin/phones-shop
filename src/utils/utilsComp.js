import { random, lorem, commerce, image } from 'faker';


export const itemStatic = {
	id: '1',
	categoryId: '1',
	name: 'Apple iPhone 5c',
	description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta',
	price: 823,
	image: '/uploads/iphone5c-selection-hero-2013.png',
	cpu: '1.3GHz Apple A6',
	camera: '8mp (3264x2448)',
	size: '124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)',
	weight: '132 grams (4.7 ounces) with battery',
	display: '4.0 326 pixel density',
	battery: '1480 mAh',
	memory: '16GB, 32GB and RAM 1 GB'
};
