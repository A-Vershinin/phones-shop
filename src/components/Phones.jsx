import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PageTemplate from './PageTemplate';


class Phones extends PureComponent {
  render() {
    return (
      <PageTemplate>
        <h1>Phones components</h1>
        <Link to='/'>Back to home</Link>
      </PageTemplate>
    );
  }

}

export default Phones;
