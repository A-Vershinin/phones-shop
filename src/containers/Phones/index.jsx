import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Phones from '../../components/Phones.jsx';
import { fetchPhonesAction, loadMorePhonesAction }  from '../../actions/phonesAction';
import { addPhoneToBasketAction } from '../../actions/basketAction';
import { fetchCategoriesAction } from '../../actions//categoriesAction.js';
import { getPhones } from '../../selectors/selectors';

class PhonesContainer extends PureComponent {
  componentWillMount() {
    this.props.fetchPhones();
    this.props.fetchCategories();
  }

  render() {
    return (<Phones {...this.props}/>);
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    phones: getPhones(state, ownProps),
  }
}

const mapStateToDispatch = (dispatch) => {
  return {
    fetchPhones: () => dispatch(fetchPhonesAction()),
    fetchCategories: () => dispatch(fetchCategoriesAction()),
    loadMorePhones: () => dispatch(loadMorePhonesAction()),
    addPhoneToBasket: (id) => dispatch(addPhoneToBasketAction(id)),
  };
};

export default withRouter(connect(mapStateToProps, mapStateToDispatch)(PhonesContainer));
