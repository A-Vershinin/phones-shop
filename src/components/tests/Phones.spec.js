import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom'
import chai, { assert, should } from 'chai';
import enzyme, { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'; // +for snapshot
import { Provider } from 'react-redux';
import chaiJestSnapshot from "chai-jest-snapshot";
import Adapter from 'enzyme-adapter-react-16';

import { random, lorem, commerce, image } from 'faker';

import PageTemplate from '../PageTemplate';
import Phones from '../Phones.jsx';
import configureStore from '../../store';
import { itemStatic } from '../../utils/utilsComp';

const expectChai = chai.expect;
enzyme.configure({ adapter: new Adapter() });

const store = configureStore({});

chai.use(chaiJestSnapshot); // for chaiJestSnapshot

describe.only('<Phones> dump', function () {
		let wrapper;
	  const phones = [itemStatic];
		const loadMorePhones = [itemStatic];

		beforeEach(() => {
			wrapper = shallow(<Phones phones={phones} loadMorePhones={loadMorePhones}/>)
		});

		it('should render DUMB component', () => {
			expectChai(wrapper.length).to.equal(1);
	 	});

		// it('should have props for phones and loadMorePhones', function () {
		// 	const wrapperD = shallow(<Phones />);
		// 	expectChai(wrapperD.props().loadMorePhones).to.be.defined;
		// 	expectChai(wrapperD.props().phones).to.be.defined;
		// });


		it.skip('snapshot of Phones with jest', () => {
			const renderer2 = new ShallowRenderer();//+
			const renderedValue = renderer2.render(<Phones phones={phones} loadMorePhones={loadMorePhones}/>);
			expect(renderer2.getRenderOutput()).toMatchSnapshot();
	 	});

		beforeEach(function() {
			chaiJestSnapshot.setFilename(__filename + ".snap");
			chaiJestSnapshot.setTestName("Phones renders correctly");
		});

		it("snapshot of Phones with mocha", () => {
			const renderer2 = new ShallowRenderer();
			const renderedValue = renderer2.render(<Phones phones={phones} loadMorePhones={loadMorePhones}/>);
	    expectChai(renderedValue).to.matchSnapshot();
	  });

});
