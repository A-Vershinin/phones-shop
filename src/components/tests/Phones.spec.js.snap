// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders dasdcorrectly 1`] = `
<PageTemplate>
  <h1>
    Popular phones
  </h1>
  <div
    className="books row"
  >
    <div
      className="col-xs-6 col-sm-4 col-lg-4 col-md-4 book-list"
    >
      <div
        className="thumbnail"
      >
        <img
          alt="Apple iPhone 5c"
          className="img-thumbnail"
          src="/uploads/iphone5c-selection-hero-2013.png"
        />
        <div
          className="caption"
        >
          <h4
            className="pull-right"
          >
            $
            823
          </h4>
          <h4>
            <Link
              replace={false}
              to="/phones/1"
            >
              Apple iPhone 5c
            </Link>
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio,...
          </p>
          <p
            className="itemButton"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Buy Now!
            </button>
            <Link
              className="btn btn-default"
              replace={false}
              to="/phones/1"
            >
              More info
            </Link>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <button
        className="pull-right btn btn-primary"
        onClick={
          Array [
            Object {
              "battery": "1480 mAh",
              "camera": "8mp (3264x2448)",
              "categoryId": "1",
              "cpu": "1.3GHz Apple A6",
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
              "display": "4.0 326 pixel density",
              "id": "1",
              "image": "/uploads/iphone5c-selection-hero-2013.png",
              "memory": "16GB, 32GB and RAM 1 GB",
              "name": "Apple iPhone 5c",
              "price": 823,
              "size": "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
              "weight": "132 grams (4.7 ounces) with battery",
            },
          ]
        }
      >
        Load More
      </button>
    </div>
  </div>
</PageTemplate>
`;

exports[`Phones renders correctly 1`] = `
<PageTemplate>
  <h1>
    Popular phones
  </h1>
  <div
    className="books row"
  >
    <div
      className="col-xs-6 col-sm-4 col-lg-4 col-md-4 book-list"
    >
      <div
        className="thumbnail"
      >
        <img
          alt="Apple iPhone 5c"
          className="img-thumbnail"
          src="/uploads/iphone5c-selection-hero-2013.png"
        />
        <div
          className="caption"
        >
          <h4
            className="pull-right"
          >
            $
            823
          </h4>
          <h4>
            <Link
              replace={false}
              to="/phones/1"
            >
              Apple iPhone 5c
            </Link>
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio,...
          </p>
          <p
            className="itemButton"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Buy Now!
            </button>
            <Link
              className="btn btn-default"
              replace={false}
              to="/phones/1"
            >
              More info
            </Link>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <button
        className="pull-right btn btn-primary"
        onClick={
          Array [
            Object {
              "battery": "1480 mAh",
              "camera": "8mp (3264x2448)",
              "categoryId": "1",
              "cpu": "1.3GHz Apple A6",
              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
              "display": "4.0 326 pixel density",
              "id": "1",
              "image": "/uploads/iphone5c-selection-hero-2013.png",
              "memory": "16GB, 32GB and RAM 1 GB",
              "name": "Apple iPhone 5c",
              "price": 823,
              "size": "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
              "weight": "132 grams (4.7 ounces) with battery",
            },
          ]
        }
      >
        Load More
      </button>
    </div>
  </div>
</PageTemplate>
`;
